### PR TITLE
Support dataset in Honeycomb propagation headers

### PR DESCRIPTION
**Pull request by martin308**
Adds support for reading the dataset from the honeycomb propagation headers and setting the dataset for the trace appropriately.

### CODE REVIEW

File: beeline/propagation/honeycomb.py [hunk 1]
```diff
@@ -43,19 +44,31 @@ def marshal_propagation_context(propagation_context):
...
     # some automatically added trace fields that we DON'T want to propagate - e.g. request.*
     version = 1
+    dataset = None
+    if propagation_context.dataset:
+        dataset = quote(propagation_context.dataset)
     trace_fields = base64.b64encode(json.dumps(
         propagation_context.trace_fields).encode()).decode()
-    trace_header = "{};trace_id={},parent_id={},context={}".format(
-        version, propagation_context.trace_id, propagation_context.parent_id, trace_fields
+    trace_header = "{};dataset={},trace_id={},parent_id={},context={}".format(
```
Comment (MikeGoldsmith) [inline beeline/propagation/honeycomb.py:52]: If dataset is `None`, we would end up with the header value of `1;dataset=,trace_id=...` - is it okay to have an empty dataset value?

File: beeline/propagation/honeycomb.py [hunk 2]
```diff
@@ -78,9 +91,11 @@ def unmarshal_propagation_context(trace_header):
             parent_id = v
         elif k == 'context':
             context = json.loads(base64.b64decode(v.encode()).decode())
+        elif k == 'dataset':
+            dataset = unquote(v)
```
Comment (MikeGoldsmith) [inline beeline/propagation/honeycomb.py:95]: Should we validate dataset has a value before calling `unquote`?

**Review by MikeGoldsmith — COMMENTED** [review]
Generally looks really good. Just a couple of questions regarding empty/`None` dataset handling.

File: beeline/propagation/honeycomb.py (hunk 2 above)
Comment (martin308) [reply beeline/propagation/honeycomb.py:95]: I'm pretty sure once we get to here that v will be a string, either empty or containing a dataset. Which either way we are safe to call unquote

File: beeline/propagation/honeycomb.py (hunk 1 above)
Comment (martin308) [reply beeline/propagation/honeycomb.py:52]: I'll see how the other beelines handle this. If you don't have a dataset then the beeline won't be sending any events

Comment (martin308) [reply beeline/propagation/honeycomb.py:52]: looks like it's pretty variable across them so I'll add some more defensiveness here

**Review by MikeGoldsmith — APPROVED** [review]

**Pull request merged**